src/python/pants/backend/python/tasks2/select_interpreter.py

```diff
@@ -51,6 +51,9 @@ def execute(self):
     python_tgts = self.context.targets(lambda tgt: isinstance(tgt, PythonTarget))
     fs = PythonInterpreterFingerprintStrategy()
     with self.invalidated(python_tgts, fingerprint_strategy=fs) as invalidation_check:
+      if PythonSetup.global_instance().interpreter_search_paths and PythonInterpreterCache.pex_python_paths:
```

Comment:
`PythonInterpreterCache.pex_python_paths` is defined above as an instance method, here you access it as a class attribute. Need some fixup in one place or the other, but I expect you want to fix `PythonInterpreterCache.pex_python_paths` to be a memoized classmethod and call `PythonInterpreterCache.pex_python_paths()` here.

Reply:
Fixed it to be a class method call, however I tried memoizing with multiple decorators from `pants.util.memo` (`memoized`, `memoized_method`) in tandem with `@classmethod` and none were able to function properly in testing and manual runs. I would prefer to follow up in a separate PR to address this if the workaround is not clear here.